--- strace-4.7/configure.ac.bero	2013-01-16 02:53:03.294408295 +0100
+++ strace-4.7/configure.ac	2013-01-16 02:53:08.515345908 +0100
@@ -3,7 +3,7 @@ AC_PREREQ(2.57)
 AC_INIT([strace],[4.7])
 AC_CONFIG_SRCDIR([strace.c])
 AC_CONFIG_AUX_DIR([.])
-AM_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADERS([config.h])
 AM_INIT_AUTOMAKE([foreign check-news dist-xz no-dist-gzip silent-rules])
 AM_MAINTAINER_MODE
 AC_CANONICAL_HOST
