--- strace-4.10/mknod.c.omv~	2015-06-23 12:31:53.619399861 +0200
+++ strace-4.10/mknod.c	2015-06-23 12:32:00.970370285 +0200
@@ -1,6 +1,7 @@
 #include "defs.h"
 
 #include <fcntl.h>
+#include <sys/stat.h>
 
 #ifdef MAJOR_IN_SYSMACROS
 # include <sys/sysmacros.h>
--- strace-4.10/printmode.c.omv~	2015-06-23 12:36:04.721428552 +0200
+++ strace-4.10/printmode.c	2015-06-23 12:36:10.949405329 +0200
@@ -1,6 +1,7 @@
 #include "defs.h"
 
 #include <fcntl.h>
+#include <sys/stat.h>
 
 #include "xlat/modetypes.h"
 
