diff -up strace-5.2/defs.h.omv~ strace-5.2/defs.h
--- strace-5.2/defs.h.omv~	2019-08-23 19:19:35.420792591 +0000
+++ strace-5.2/defs.h	2019-08-23 19:24:36.973224735 +0000
@@ -44,6 +44,11 @@
 # include "sysent.h"
 # include "xmalloc.h"
 
+#ifdef __arm__
+// FIXME evil hack to make stuff in -m32 compile
+typedef __uint64_t __uint128_t;
+#endif
+
 # ifndef HAVE_STRERROR
 const char *strerror(int);
 # endif
diff -up strace-5.2/xlat/v4l2_pix_fmts.h.omv~ strace-5.2/xlat/v4l2_pix_fmts.h
--- strace-5.2/xlat/v4l2_pix_fmts.h.omv~	2019-08-23 19:41:11.001345160 +0000
+++ strace-5.2/xlat/v4l2_pix_fmts.h	2019-08-23 19:42:16.015947809 +0000
@@ -271,10 +271,10 @@ DIAG_POP_IGNORE_TAUTOLOGICAL_COMPARE
 #endif
 #if defined(V4L2_PIX_FMT_BGRA444) || (defined(HAVE_DECL_V4L2_PIX_FMT_BGRA444) && HAVE_DECL_V4L2_PIX_FMT_BGRA444)
 DIAG_PUSH_IGNORE_TAUTOLOGICAL_COMPARE
-static_assert((V4L2_PIX_FMT_BGRA444) == (v4l2_fourcc('B', 'A', '1', '2')), "V4L2_PIX_FMT_BGRA444 != v4l2_fourcc('B', 'A', '1', '2')");
+static_assert((V4L2_PIX_FMT_BGRA444) == (v4l2_fourcc('G', 'A', '1', '2')), "V4L2_PIX_FMT_BGRA444 != v4l2_fourcc('G', 'A', '1', '2')");
 DIAG_POP_IGNORE_TAUTOLOGICAL_COMPARE
 #else
-# define V4L2_PIX_FMT_BGRA444 v4l2_fourcc('B', 'A', '1', '2')
+# define V4L2_PIX_FMT_BGRA444 v4l2_fourcc('G', 'A', '1', '2')
 #endif
 #if defined(V4L2_PIX_FMT_SGRBG12) || (defined(HAVE_DECL_V4L2_PIX_FMT_SGRBG12) && HAVE_DECL_V4L2_PIX_FMT_SGRBG12)
 DIAG_PUSH_IGNORE_TAUTOLOGICAL_COMPARE
