diff -up strace-5.2/defs.h.omv~ strace-5.2/defs.h
--- strace-5.2/defs.h.omv~	2019-08-23 19:19:35.420792591 +0000
+++ strace-5.2/defs.h	2019-08-23 19:24:36.973224735 +0000
@@ -44,6 +44,11 @@
 # include "sysent.h"
 # include "xmalloc.h"
 
+#ifdef __arm__
+// FIXME evil hack to make stuff in -m32 compile
+typedef __uint64_t __uint128_t;
+#endif
+
 # ifndef HAVE_STRERROR
 const char *strerror(int);
 # endif
diff -up strace-5.2/linux/aarch64/arch_regs.c.omv~ strace-5.2/linux/aarch64/arch_regs.c
--- strace-5.2/linux/aarch64/arch_regs.c.omv~	2019-08-23 19:38:03.265875640 +0000
+++ strace-5.2/linux/aarch64/arch_regs.c	2019-08-23 19:40:48.092429816 +0000
@@ -27,6 +27,13 @@ struct arm_pt_regs {
 #define ARM_r0		uregs[0]
 #define ARM_ORIG_r0	uregs[17]
 
+struct user_pt_regs {
+        uint64_t        regs[31];
+        uint64_t        sp;
+        uint64_t        pc;
+        uint64_t        pstate;
+};
+
 static union {
 	struct user_pt_regs aarch64_r;
 	struct arm_pt_regs  arm_r;
diff -up strace-5.2/mpers-m32/siginfo_t.c.omv~ strace-5.2/mpers-m32/siginfo_t.c
diff -up strace-5.2/mpers.sh.omv~ strace-5.2/mpers.sh
diff -up strace-5.2/xlat/open_mode_flags.h.omv~ strace-5.2/xlat/open_mode_flags.h
--- strace-5.2/xlat/open_mode_flags.h.omv~	2019-08-23 19:29:24.417067315 +0000
+++ strace-5.2/xlat/open_mode_flags.h	2019-08-23 19:34:59.274643442 +0000
@@ -296,10 +296,10 @@ DIAG_POP_IGNORE_TAUTOLOGICAL_COMPARE
 #elif defined __arm__ || defined __arm64__ || defined __aarch64__ || defined __m68k__
 #if defined(O_DIRECT) || (defined(HAVE_DECL_O_DIRECT) && HAVE_DECL_O_DIRECT)
 DIAG_PUSH_IGNORE_TAUTOLOGICAL_COMPARE
-static_assert((O_DIRECT) == (0200000), "O_DIRECT != 0200000");
+static_assert((O_DIRECT) == (040000), "O_DIRECT != 040000");
 DIAG_POP_IGNORE_TAUTOLOGICAL_COMPARE
 #else
-# define O_DIRECT 0200000
+# define O_DIRECT 040000
 #endif
 #elif defined __powerpc__
 #if defined(O_DIRECT) || (defined(HAVE_DECL_O_DIRECT) && HAVE_DECL_O_DIRECT)
@@ -353,10 +353,10 @@ DIAG_POP_IGNORE_TAUTOLOGICAL_COMPARE
 #elif defined __alpha__ || defined __arm__ || defined __arm64__ || defined __aarch64__ || defined __m68k__
 #if defined(O_LARGEFILE) || (defined(HAVE_DECL_O_LARGEFILE) && HAVE_DECL_O_LARGEFILE)
 DIAG_PUSH_IGNORE_TAUTOLOGICAL_COMPARE
-static_assert((O_LARGEFILE) == (0400000), "O_LARGEFILE != 0400000");
+static_assert((O_LARGEFILE) == (0100000), "O_LARGEFILE != 0100000");
 DIAG_POP_IGNORE_TAUTOLOGICAL_COMPARE
 #else
-# define O_LARGEFILE 0400000
+# define O_LARGEFILE 0100000
 #endif
 #elif defined __sparc__
 #if defined(O_LARGEFILE) || (defined(HAVE_DECL_O_LARGEFILE) && HAVE_DECL_O_LARGEFILE)
@@ -386,10 +386,10 @@ DIAG_POP_IGNORE_TAUTOLOGICAL_COMPARE
 #elif defined __arm__ || defined __arm64__ || defined __aarch64__ || defined __m68k__ || defined __powerpc__
 #if defined(O_NOFOLLOW) || (defined(HAVE_DECL_O_NOFOLLOW) && HAVE_DECL_O_NOFOLLOW)
 DIAG_PUSH_IGNORE_TAUTOLOGICAL_COMPARE
-static_assert((O_NOFOLLOW) == (0100000), "O_NOFOLLOW != 0100000");
+static_assert((O_NOFOLLOW) == (0400000), "O_NOFOLLOW != 0400000");
 DIAG_POP_IGNORE_TAUTOLOGICAL_COMPARE
 #else
-# define O_NOFOLLOW 0100000
+# define O_NOFOLLOW 0400000
 #endif
 #elif defined __alpha__
 #if defined(O_NOFOLLOW) || (defined(HAVE_DECL_O_NOFOLLOW) && HAVE_DECL_O_NOFOLLOW)
@@ -535,10 +535,10 @@ DIAG_POP_IGNORE_TAUTOLOGICAL_COMPARE
 #elif defined __arm__ || defined __arm64__ || defined __aarch64__ || defined __m68k__ || defined __powerpc__
 #if defined(O_DIRECTORY) || (defined(HAVE_DECL_O_DIRECTORY) && HAVE_DECL_O_DIRECTORY)
 DIAG_PUSH_IGNORE_TAUTOLOGICAL_COMPARE
-static_assert((O_DIRECTORY) == (040000), "O_DIRECTORY != 040000");
+static_assert((O_DIRECTORY) == (0200000), "O_DIRECTORY != 0200000");
 DIAG_POP_IGNORE_TAUTOLOGICAL_COMPARE
 #else
-# define O_DIRECTORY 040000
+# define O_DIRECTORY 0200000
 #endif
 #elif defined __alpha__
 #if defined(O_DIRECTORY) || (defined(HAVE_DECL_O_DIRECTORY) && HAVE_DECL_O_DIRECTORY)
diff -up strace-5.2/xlat/v4l2_pix_fmts.h.omv~ strace-5.2/xlat/v4l2_pix_fmts.h
--- strace-5.2/xlat/v4l2_pix_fmts.h.omv~	2019-08-23 19:41:11.001345160 +0000
+++ strace-5.2/xlat/v4l2_pix_fmts.h	2019-08-23 19:42:16.015947809 +0000
@@ -271,10 +271,10 @@ DIAG_POP_IGNORE_TAUTOLOGICAL_COMPARE
 #endif
 #if defined(V4L2_PIX_FMT_BGRA444) || (defined(HAVE_DECL_V4L2_PIX_FMT_BGRA444) && HAVE_DECL_V4L2_PIX_FMT_BGRA444)
 DIAG_PUSH_IGNORE_TAUTOLOGICAL_COMPARE
-static_assert((V4L2_PIX_FMT_BGRA444) == (v4l2_fourcc('B', 'A', '1', '2')), "V4L2_PIX_FMT_BGRA444 != v4l2_fourcc('B', 'A', '1', '2')");
+static_assert((V4L2_PIX_FMT_BGRA444) == (v4l2_fourcc('G', 'A', '1', '2')), "V4L2_PIX_FMT_BGRA444 != v4l2_fourcc('G', 'A', '1', '2')");
 DIAG_POP_IGNORE_TAUTOLOGICAL_COMPARE
 #else
-# define V4L2_PIX_FMT_BGRA444 v4l2_fourcc('B', 'A', '1', '2')
+# define V4L2_PIX_FMT_BGRA444 v4l2_fourcc('G', 'A', '1', '2')
 #endif
 #if defined(V4L2_PIX_FMT_SGRBG12) || (defined(HAVE_DECL_V4L2_PIX_FMT_SGRBG12) && HAVE_DECL_V4L2_PIX_FMT_SGRBG12)
 DIAG_PUSH_IGNORE_TAUTOLOGICAL_COMPARE
